geomfn: prevent smaller densifyFrac for ST_{Frechet,Hausdorff}Distance

otan opened the pull request:
Release justification: low-risk change to existing functionality

Resolves #61367.

Release note (sql change): Prevent densifyFracs < 1e-6 for
ST_FrechetDistance and ST_HausdorffDistance to protect panics and out of
memory errors.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61427)
<!-- Reviewable:end -->

rafiss reviewed (approved):
comment fix then lgtm!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/61427#-:-MUttoHnCkaMu8n_MHZ6:b-ggsrid)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/geomfn/distance_test.go, line 795 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/61427#-MUttbX25qUqAPC2r715:-MUttbX25qUqAPC2r716:b-giuckc) ([raw file](https://github.com/cockroachdb/cockroach/blob/e85eea9097f7dc6931abbdc916e444297b0eef7e/pkg/geo/geomfn/distance_test.go#L795)):*
> ```Go
> 		// The threshold was empirically found to be at 1e-20, while at 1e-19
> 		// GEOS instead returns the error "geos error: vector". We explicitly
> 		// disallow <1e-19 in the code, and test that both of these error. We do
> ```

nit: comment needs updating


<!-- Sent from Reviewable.io -->

otan committed:
geomfn: prevent smaller densifyFrac for ST_{Frechet,Hausdorff}Distance

Release justification: low-risk change to existing functionality

Release note (sql change): Prevent densifyFracs < 1e-6 for
ST_FrechetDistance and ST_HausdorffDistance to protect panics and out of
memory errors.

otan commented:
bors r=rafiss
